crates/hotshot/src/traits/networking/libp2p_network.rs

```diff
@@ -107,6 +109,8 @@ struct Libp2pNetworkInner<M: NetworkMsg, K: SignatureKey + 'static> {
     /// hash(hashset) -> topic
     /// btreemap ordered so is hashable
     topic_map: RwLock<BiHashMap<BTreeSet<K>, String>>,
+    /// the current view number (for node lookup purposes)
+    cur_view: Arc<AtomicU64>,
```

Comment:
We can make it generic over the ViewNumber, but then we would have to wrap it in an RwLock which I _think_ would be slower. Perhaps I can make a type like `AtomicViewNumber`?

Reply:
Yeesh this is difficult. I think the assumption for AtomicViewNumber to work is that it can be represented as a series of atomicusizes. I'm not sure if that's true in practice. I guess we could do `Arc<Mutex<ViewNumber>>` but lockfree is definitely nicer.

Maybe this is the best we can do? It's probably worthwhile to add a note explicitly saying that doing this generically over viewnumber is hard so we convert it into a usize.